View item details

sar-mko opened the pull request:
## Description

- In `ListItem` created a div to hold item details
- Passed in `dateNextPurchased` and `totalPurchases` from the `item` object
- displayed the details in a list
- created a toggle button to switch between open and closed states

Ideally, we can use this as a base for more stylistic choices coming up.

## Related Issue

closes #37 (#21)

## Acceptance Criteria

- [x] User has the option to view details on an item from their list
- [x] User can see `dateLastPurchased`, `dateNextPurchase`, and `totalPurchases` pertaining to a single item.
- [x] Data is accurate to the specific item

## Type of Changes

`enhancement`

## Updates

### After
<img width="461" alt="Screenshot 2024-09-24 at 11 51 50 AM" src="https://github.com/user-attachments/assets/e656e8a6-8794-4e37-a175-e82fa8085d54">

<img width="496" alt="Screenshot 2024-09-24 at 11 51 59 AM" src="https://github.com/user-attachments/assets/fe5e79a2-3308-4274-bdfb-db7e504a2a0d">

## Testing Steps / QA Criteria

- View list of items on `List` and there should be a `View Purchase Details` button
- Clicking button will open and should display details
- Open Firebase to list info see if `dateLastPurchased`, `dateNextPurchase`, and `totalPurchases` are the same as whats displayed
- Clicking button should toggle between displaying and hiding the div

arandel1 reviewed (commented):
Functionality works great! 

**nit-pick/non-blocking:** To set us up for adding CSS later, can we make the list items as `<p>` instead of `<h2>`? They just look a little big.

**nit-pick/non-blocking:** In the View Details section, can we add a space between the three stats? They currently look like this: "Last Purchase:Fri Sep 20 2024" and I'm wanting a space after the colon: "Last Purchase: Fri Sep 20 2024".

EmmaBin reviewed (approved):
I tested by adding a new item and making purchases, and all data was correctly reflected in the div.

sar-mko commented:
> Functionality works great!
> 
> **nit-pick/non-blocking:** To set us up for adding CSS later, can we make the list items as `<p>` instead of `<h2>`? They just look a little big.
> 
> **nit-pick/non-blocking:** In the View Details section, can we add a space between the three stats? They currently look like this: "Last Purchase:Fri Sep 20 2024" and I'm wanting a space after the colon: "Last Purchase: Fri Sep 20 2024".

Hi, I added fontSize inline for now and added a span around the date to get the spacing.

MarcosPerez16 reviewed (approved):
Tested and it everything is working on my end.
